topdown/http.go

```diff
@@ -229,6 +231,41 @@ func canonicalizeHeaders(headers map[string]interface{}) map[string]interface{}
 	return canonicalized
 }
 
+// useSocket examines the url for "unix://" and returns a *http.Transport with
+// a DialContext that opens a socket (specified in the http call).
+// The url is expected to contain socket=/path/to/socket (url encoded)
+// Ex. "unix:localhost/end/point?socket=%2Ftmp%2Fhttp.sock"
+func useSocket(rawURL string, tlsConfig *tls.Config) (bool, string, *http.Transport) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return false, rawURL, nil
+	}
+
+	if u.Scheme == "unix" {
+		// Get the path to the socket
+		if u.RawQuery != "" {
```

Comment:
[nit] We could reduce the indentation here if we weed out the bad inputs first, like
```go
if u.Scheme != "unix" || u.RawQuery == "" {
	return false, rawURL, nil
}

v, err := url.ParseQuery(u.RawQuery)
// ... etc
```

Reply:
Also, I believe we never use `rawURL` if the `ok` return isn't true, so we could indicate that here as well,
```go
return false, "", nil
```
instead of passing `rawURL` back out.